src/substrate/substrate.handler.ts

```diff
@@ -458,6 +470,7 @@ export class SubstrateService
     if (this.listenStatus) {
       this.logger.log('Restart connection to node.');
       this.stopListen();
+      await this.setWsProvider();
       await this.startListen();
     }
```

Comment:
too many codes inside this `if` statement, please make it simpler by using the return instead